extras/CatchAddTests.cmake

```diff
@@ -16,7 +16,10 @@ set(tests)
 
 function(add_command NAME)
   set(_args "")
-  foreach(_arg ${ARGN})
+  # use ARGV* instead of ARGN, because ARGN splits arrays into multiple arguments
+  math(EXPR _last_arg ${ARGC}-1)
+  foreach(_n RANGE 1 ${_last_arg})
```

Comment:
Wouldn't replacing the original `foreach` with `foreach (arg IN LISTS ARGV)` work too?

Reply:
You would think so, but is actually doesn't work. `ARGV` and `ARGN` split arguments that contain semicolons into multiple entries, as demonstrated by the following cmake script:

```cmake
function(test_arguments)
	message("using \"IN LISTS ARGV\" (incorrect):")
	foreach(arg IN LISTS ARGV)
		message("  arg=${arg}")
	endforeach()

	message("using \"ARGV0..ARGVn\" (correct):")
	math(EXPR last_arg ${ARGC}-1)
	foreach(n RANGE 0 ${last_arg})
		set(arg "${ARGV${n}}")
		message("  arg=${arg}")
	endforeach()
endfunction(test_arguments)

set (arguments "one;two\;three;four")

test_arguments(${arguments })
```
which outputs:

```
using "IN LISTS ARGV" (incorrect):
  arg=one
  arg=two
  arg=three
  arg=four
using "ARGV0..ARGVn" (correct):
  arg=one
  arg=two;three
  arg=four
```

The second output is what we need. I originally thought `foreach(_arg IN LISTS ARGN)` would word as well, but no such luck.

Reply:
Thanks for testing it, also goddamit CMake.